src/main/java/co/cask/gcp/gcs/GCSConfigHelper.java

```diff
@@ -36,12 +36,9 @@
   public static URI getPath(String path) {
     try {
       URI uri = new URI(path);
-      if (uri.getScheme() != null && !uri.getScheme().equalsIgnoreCase("gs")) {
-        throw new URISyntaxException(path, String.format("Invalid scheme. Scheme must be '%s'.",
-                                                         FILE_SYSTEM_PATH_PREFIX));
-      }
-      if (uri.getScheme() != null && uri.getAuthority() == null) {
-        throw new URISyntaxException(path, "Invalid bucket name in path.");
+      if (uri.getScheme() != null && (!uri.getScheme().equalsIgnoreCase("gs") || uri.getAuthority() == null)) {
+        throw new URISyntaxException(path, String.format("Invalid path '%s'. The path must be of form " +
+                                                           "'gs://<bucket-name>/path'.", FILE_SYSTEM_PATH_PREFIX));
```

Comment:
wrong substitution

Reply:
also if you throw this here, it will be caught by the outer catch and the message will be lost. Should use this message if the path can't be parsed at all as well as here.

Reply:
That is what I was trying to achieve. Since I include the exception in the illegal argument exception it will preserve the message.

Reply:
It's included as the cause. Which means the message will be different. The message is what we care most about, it is what shows up in the UI.

Reply:
speaking of which, can you make sure this is called during validation